lib/promise-all.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
```

Comment:
What does this file have to do with the `AsyncIterator` implementation?

Reply:
@nechaido said it should be in this PR, but I also think that this is a separate functionality. It can be used in an iterator for some methods, such as 'every'.